b = 10
def f(a):
    print(a)
    print(b)
    b = 15

f(3)


Произойдёт ошибка: переменная b используется внутри функции
f перед ее присвоением значения внутри функции.